### PR TITLE
DAOS-6613 dfs: Improve error handling in dfs_open()

### PR DESCRIPTION
**Pull request by ashleypittman**
Avoid a potential leak on the error path when dfs_open fails.
Improve the shutdown order in dfuse_loop() to ensure semaphore
is correctly destroyed.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
Other than unrelated NLT errors this passed CI so can be reviewed.  I'll update or merge with master depending on feedback but putting it into the queue now wouldn't help.

**Review by daltonbohning — APPROVED** [review]
LGTM.

**Comment by dsikich** [comment]
> LGTM.

@ashleypittman the changes seem pretty straightforward with a lot of them fixing spacing/alignment and removing the if check for D_FREE. I can approve or hold off if you want @mchaarawi to review/approve instead.

**Comment by ashleypittman** [comment]
> > LGTM.
> 
> @ashleypittman the changes seem pretty straightforward with a lot of them fixing spacing/alignment and removing the if check for D_FREE. I can approve or hold off if you want @mchaarawi to review/approve instead.

If you could approve it that would be great.

**Review by mchaarawi — APPROVED** [review]

File: src/client/dfs/dfs.c
Comment (ashleypittman) [reply src/client/dfs/dfs.c:572]: I know, there's no real way to do it nicely with the trigraphs there.

Comment (daltonbohning) [reply src/client/dfs/dfs.c:572]: Personally, I prefer this style:
```
call_to_some_fun(
	first_arg_indented,
	second_arg,
	...)
```
Then long function names don't cause this issue

Comment (ashleypittman) [reply src/client/dfs/dfs.c:572]: We have a CI check for parameter alignment that checks things are aligned, however long function names and the 80 character limit make it very difficult to do somethings.  Properly aligned normally helps readability but in this case I agree it doesn't.

I've been working in a git hook to detect and fix a number of issues before they even get pushed and overall it's working well, this is an unfortunate side-effect of that.  You'll see that several of my PRs from recent weeks have miscellaneous whitespace changes in, although it tends to only happen once per file as it's programmatic.

**Review by dsikich — APPROVED** [review]

**Pull request merged**
